Feature: If RSS is enabled, show RSS icon in menu

vvzen opened the pull request:
This PR adds support for showing a tiny RSS svg badge in the top menu in case the user added a `menu_item` that looks like this:

```toml
menu_items = [
    {name = "home", url = "$BASE_URL"},

    # [...]

    # RSS
    {name = "", url = "$BASE_URL/$FEED_FILENAME"},
]
```
    
It also requires that the user has a config.toml that contains the following entries, and `generate_feed` is set to `true` :
    
```toml
generate_feed = true
feed_filename = "rss.xml"
author = "Your name here"
```

The icon would look like this:
![Screenshot 2024-06-01 at 11 51 14 AM](https://github.com/pawroman/zola-theme-terminimal/assets/10340139/eb2ec094-a28a-4d2b-bab3-fb3c8932689f)

A live example can be seen here: https://valerioviperino.me

vvzen commented:
Oh, I didn't know about that OSS icon, that's much better!

As for the other things, sounds good! I'll have a look as soon as I have a bit of free time.

vvzen commented:
If I use something like
```svg
<svg xmlns="http://www.w3.org/2000/svg" width="22" height="22" viewBox="0 0 32 32" fill="none" stroke="#ee802f" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-rss">
<g transform="translate(0,5)">
    <path d="M4 11a9 9 0 0 1 9 9"></path>
    <path d="M4 4a16 16 0 0 1 16 16"></path>
    <circle cx="5" cy="19" r="1"></circle>
</g>
</svg>
```

it looks like this:
![Screenshot 2024-06-06 at 10 39 11 PM](https://github.com/pawroman/zola-theme-terminimal/assets/10340139/4c686425-1966-4792-958a-d9fc720ba6f0)

is that okay?

Otherwise I can keep it white and slightly bigger, which is what your original snippet does!

vvzen commented:
As for enabling the icon, right now one would need to add this to the `menu_items` :
```toml
menu_items = [
    # RSS
    {name = "", url = "$BASE_URL/$FEED_FILENAME"}
]
```

so it's opt-in already, in a way. Unless you prefer a dedicated config entry!
As for the color, it should be pretty easy to make it configurable and just inject the `rss_icon_color` as a variable into the svg itself.

pawroman commented:
> is that okay?

yeah, nice!

3 things I would add on top of it (in order of priority):

1. The `svg` element should have a CSS class or id so that it's easy to customize the size, and other things.
2. It would also be great to document how to add the RSS feed icon in the README. E.g. it's not obvious that menu items with `url = "$BASE_URL/$FEED_FILENAME"` are going to be converted into a neat RSS icon in the menu.
3. It would be great to add `rss_icon_color` as a configurable, so it's easy to change without involving custom CSS.

vvzen commented:
I've done a first pass of those features!
1 - > d451ee0b1eee1e28fd7548019c9cbcbddc656683, I've used `id="rss-icon"`
2 -> f1b4c1de4c8616402f384034939ca63ae101dfbc
3 -> 2c4cf6f1fee08989bfbec5e1557b0269c7cf65f1, added it to the `extra` section of the `config.toml`

For 3) I think it might make more sense to have `rss_icon_color` to use the same logic that you're using for `accent_color` and `background_color`, but I need to take a deeper look at how you've implemented those! It would make it easier for people to just have the RSS icon match the current palette of the theme.

Let me know what's next!

vvzen committed:
feat: Show RSS icon if config enables it

This adds support for showing a tiny RSS svg badge in the top menu in
case the user added a menu_item that looks like this:
`{name = "", url = "$BASE_URL/$FEED_FILENAME"}`

This also requires that the user has a config.toml that contains the
following entries:

generate_feed = true
feed_filename = "rss.xml"
author = "your name here"

vvzen committed:
feat: Also gate on config.generate_feed

vvzen committed:
chore: Newlines

vvzen committed:
feat: Change RSS icon, do not prepend {{name}}

There's some css rules for which prepending the name of the menu item as
set in the config causes the <a> element to appear shifted down compared
to the other <a> elements of the menu. I might have a look later, but
for now this enables only the appeareance of the RSS icon, discarding
entirely the name of the link.

vvzen committed:
feat: Support atom rss too

vvzen committed:
feat: Initial support for custom rss icon color

At some point this might follow the same logic used by `accent_color`
and `background_color`, for consistency.

vvzen committed:
docs: Add entry documenting RSS icon settings

pawroman commented on templates/macros/menu.html:
```diff
@@ -38,11 +38,43 @@
+                            {% endif -%}
+
+                            <a type="application/{{feed_type}}" title="RSS" href="{{ get_url(path=config.feed_filename) | safe }}">
+                                <svg id="rss-icon" xmlns="http://www.w3.org/2000/svg" width="22" height="22" viewBox="0 0 32 32" fill="none" stroke="{{rss_icon_color}}" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-rss">
```
We should give credit to feather icons https://feathericons.com/ and include their LICENSE file in the repo https://github.com/feathericons/feather/blob/main/LICENSE as `LICENSE-Feather.md` (similar to Hack fonts).

vvzen replied:
Added in https://github.com/pawroman/zola-theme-terminimal/pull/63/commits/5017bf67a0ad577cd6fad6ffe05701e783583ec5

pawroman commented on README.md:
````diff
@@ -346,6 +346,41 @@ page_titles = "combined"
+To enable RSS you need to set those 3 configuration settings to valid values: 
+
+```toml
+generate_feed = true
+author = "yourself@email.com"
+
+# Use `rss.xml` for RSS feeds and `atom.xml` for ATOM.
+feed_filename = "atom.xml"
````
This needs to be updated to reflect changes introduced in #71 and #73

pawroman commented on templates/macros/menu.html:
```diff
@@ -38,11 +38,43 @@
-                        <a href="{{ item.url | replace(from="$BASE_URL", to=config.base_url) | safe }}" target="_blank" rel="noopener noreferrer">{{ item.name | safe }}</a>
+
+                    <!-- RSS -->
+                    {%- set is_rss = item.url == "$BASE_URL/$FEED_FILENAME" -%}
```
We use

`{%- if feed is containing('rss') %}`

elsewhere now (see #71), would be good to keep it consistent.

vvzen replied:
To do that, I'd need to do it inside the `{%- for feed in config.feed_filenames %}` loop, but this specific _if_ is guarding against a different thing: whether or not the current item refers to the RSS icon or not (we don't care about RSS or ATOM at this stage). If it is referring to RSS, then we can add the svg icon and also add the relevant `<a>` element depending if it's ATOM or RSS. Does that make sense ?

pawroman commented on templates/macros/menu.html:
```diff
@@ -38,11 +38,43 @@
+                    {%- endif %}
+
+                    {%- if is_rss -%}
+                        {%- if config.generate_feed %}
+                            {%- if config.feed_filename == "rss.xml" %}
+                                {% set feed_type = 'rss+xml' %}
+                            {%- else %}
+                                {% set feed_type = 'atom+xml' %}
+                            {% endif -%}
```
Needs to be updated, see other comments for details.

pawroman reviewed (changes requested):
Hey, thanks for addressing the comments, and sorry for the late review. Unfortunately, to merge this we need to address breaking changes to feed generation added in recent PRs. Please take a look.

vvzen commented:
Hey @pawroman , no problem. I noticed these new changes in Zola 19.1 as well (https://github.com/getzola/zola/issues/2566) - I'll give it ago once I have a bit of free time.

vvzen committed:
docs: Add License for Feather Icons

vvzen committed:
feat: Compatibility with zola >=0.19

NOTE: This breaks compat with previous previous versions since Zola made a
breaking change in 0.19 and imho it's not worth trying to support previous
versions too.
